### PR TITLE
CloudConnectionStatusService converted to Factory Component

### LINKED ISSUE #297: [Status led] What connection instance controls the status led?

### PR DESCRIPTION
**Pull request by LucaDazi**
DefaultCloudServiceFactory modified so to include CloudConnectionStatusService in stack creation
ConfigurationUpgrade changed so to update old configuration for new CloudConnectionStatusFactory Factory Component

fixes #297

### CODE REVIEW

**Comment by MMaiero** [comment]
@LucaDazi Could you please rebase the code in this branch?

File: kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/upgrade/ConfigurationUpgrade.java [hunk 1]
```diff
@@ -42,6 +44,7 @@
 	public static XmlComponentConfigurations upgrade(XmlComponentConfigurations xmlConfigs) {
 		List<ComponentConfigurationImpl> result = new ArrayList<ComponentConfigurationImpl>();
 		
+		boolean CloudConnectionStatusConfigFound = false;
```
Comment (MMaiero) [inline kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/upgrade/ConfigurationUpgrade.java]: @LucaDazi this variable should start with a lowercase character

**Comment by cdealti** [comment]
1. ``kura.build.properties`` should be updated.

It does not compile:
```
[ERROR] import org.eclipse.kura.configuration.ConfigurationService;
[ERROR] ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] Access restriction: The type ConfigurationService is not accessible due to restriction on classpath entry /home/cristiano/dev/git/kura/kura/org.eclipse.kura.api/target/classes
[ERROR] /home/cristiano/dev/git/kura/kura/org.eclipse.kura.core.status/src/main/java/org/eclipse/kura/core/status/CloudConnectionStatusServiceImpl.java:[99]
[ERROR] String pid = (String) componentContext.getProperties().get(ConfigurationService.KURA_SERVICE_PID);
[ERROR] ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ERROR] Access restriction: The type ConfigurationService is not accessible due to restriction on classpath entry /home/cristiano/dev/git/kura/kura/org.eclipse.kura.api/target/classes
[ERROR] /home/cristiano/dev/git/kura/kura/org.eclipse.kura.core.status/src/main/java/org/eclipse/kura/core/status/CloudConnectionStatusServiceImpl.java:[99]
[ERROR] String pid = (String) componentContext.getProperties().get(ConfigurationService.KURA_SERVICE_PID);
[ERROR] ^^^^^^^^^^^^^^^^
[ERROR] Access restriction: The field KURA_SERVICE_PID from the type ConfigurationService is not accessible due to restriction on classpath entry /home/cristiano/dev/git/kura/kura/org.eclipse.kura.api/target/classes
[ERROR] 3 problems (3 errors)
```

**Comment by cdealti** [comment]
@LucaDazi can we close this

**Comment by cdealti** [comment]
Closing it. Consider submitting a new one.

**Comment by cdealti** [comment]
Closing it. Consider submitting a new one.

**Pull request closed without merging**